tasmota/tasmota_xdrv_driver/xdrv_48_timeprop.ino

```diff
@@ -17,325 +17,517 @@
-        ResponseCmndNumber(newInterval);
-      }
-    }
-    else {
-      ResponseCmndNumber(maxIntervals[XdrvMailbox.index]);
-    }
+void CmndTimePropCount(void)
+{
+  if ((XdrvMailbox.payload >= 1) && (XdrvMailbox.payload <= MAX_RELAYS))
+  {
+    Timeprop.count = XdrvMailbox.payload;
+
+    Settings->timeprop_cfg.count = Timeprop.count - 1;
+    SettingsSave(0);
   }
+
+  AllocateTimepropValues();
+
+  ResponseCmndNumber(Timeprop.count);
 }
 
-void TimepropEverySecond(void) {
-  Tprop.current_time_secs++;    // increment time
-  for (int i=0; i<TIMEPROP_NUM_OUTPUTS; i++) {
-    int newState = Tprop.timeprops[i].tick(Tprop.current_time_secs);
-    if (newState != bitRead(Tprop.current_relay_states, Tprop.relay_nos[i]-1)){
-      // remove the third parameter below if using tasmota prior to v6.0.0
-      ExecuteCommandPower(Tprop.relay_nos[i], newState,SRC_IGNORE);
-    }
+void CmndTimePropLoadType(void)
+{
+  if ((XdrvMailbox.payload >= 0) && (XdrvMailbox.payload <= 1))
+  {
+    Timeprop.load_type = XdrvMailbox.payload;
+
+    Settings->timeprop_cfg.load_type = Timeprop.load_type;
+    SettingsSave(0);
   }
+
+  AllocateTimepropValues();
+
+  ResponseCmndNumber(Timeprop.load_type);
 }
 
-// called by the system each time a relay state is changed
-void TimepropXdrvPower(void) {
-  // for a single relay the state is in the lsb of index, I have think that for
-  // multiple outputs then succesive bits will hold the state but have not been
-  // able to test that
-  Tprop.current_relay_states = XdrvMailbox.index;
+void CmndTimePropFallbackAfter(void)
+{
+  if (XdrvMailbox.payload >= 0)
+  {
+
+    Settings->timeprop_cfg.fallback_time = reduceFallbackTime(XdrvMailbox.payload);
+    Timeprop.fallback_time = expandFallbackTime(Settings->timeprop_cfg.fallback_time);
+
+    SettingsSave(0);
+  }
+  ResponseCmndNumber(Timeprop.fallback_time);
 }
 
-void ShowValues(void) {
-#if TIMEPROP_REPORT_SETTINGS
-  ResponseAppend_P(PSTR(",\"Timeprop\":{"));
-  for (int i=0; i<TIMEPROP_NUM_OUTPUTS; i++) {
-    ResponseAppend_P(PSTR("\"Output%d\":{"),i);
-    ResponseAppend_P(PSTR("\"CycleTime\":%d,"),cycleTimes[i]);
-    ResponseAppend_P(PSTR("\"DeadTime\":%d,"),deadTimes[i]);
-    ResponseAppend_P(PSTR("\"OutputInvert\":%d,"),opInverts[i]);
-    ResponseAppend_P(PSTR("\"FallbackPower\":%2_f,"),&fallbacks[i]);
-    ResponseAppend_P(PSTR("\"MaxUpdateInterval\":%d"),maxIntervals[i]);
-    ResponseAppend_P(i<TIMEPROP_NUM_OUTPUTS-1 ? PSTR("},") : PSTR("}"));
+void CmndTimePropFallbackValue(void)
+{
+  if ((XdrvMailbox.payload >= 0) && (XdrvMailbox.payload <= 100))
+  {
+
+    Settings->timeprop_cfg.fallback_value = reduceFallbackValue(XdrvMailbox.payload);
+    Timeprop.fallback_value = expandFallbackValue(Settings->timeprop_cfg.fallback_value);
+
+    SettingsSave(0);
```

Comment:
Non-sense doing these SettingsSave. They are performed by the scheduler. Need to be removed